### PR TITLE
Randomly sample by constructing a key with the specified digest

### PR DESCRIPTION
**Pull request by tylerwilliams**
This changes the randomsample() function so it generates sample keys by constructing a valid PebbleKey and then serializing keyBytes at the current database version. This will allow random sampling to work correctly across database versions.

### CODE REVIEW

File: enterprise/server/backends/pebble_cache/pebble_cache.go [hunk 1]
```diff
@@ -1574,6 +1579,7 @@ func newPartitionEvictor(part disk.Partition, fileStorer filestore.Store, blobDi
 	pe.sizeBytes = sizeBytes
 	pe.casCount = casCount
 	pe.acCount = acCount
+	pe.lru.UpdateSizeBytes(sizeBytes)
```
Comment (tylerwilliams) [inline enterprise/server/backends/pebble_cache/pebble_cache.go:1582]: this is a tiny fix -- previously we were not updating the lru size until a write happened -- now we also update it when reading the cached size from the db (or if it was force recomputed).

File: enterprise/server/backends/pebble_cache/pebble_cache.go [hunk 2]
```diff
@@ -165,8 +165,7 @@ type PebbleCache struct {
 	atimeBufferSize      int
 	minEvictionAge       time.Duration
 
-	lastDBVersionUpdate time.Time
-	lastDBVersion       filestore.PebbleKeyVersion
+	lastDBVersion filestore.PebbleKeyVersion
```
Comment (tylerwilliams) [inline enterprise/server/backends/pebble_cache/pebble_cache.go:168]: The lastDBVersionUpdate was a vestige of something removed in a previous CL and unused, so I removed it.

File: enterprise/server/backends/pebble_cache/pebble_cache.go [hunk 3]
```diff
@@ -378,7 +377,7 @@ func NewPebbleCache(env environment.Env, opts *Options) (*PebbleCache, error) {
 			if err := disk.EnsureDirectoryExists(blobDir); err != nil {
 				return err
 			}
-			pe, err := newPartitionEvictor(part, pc.fileStorer, blobDir, pc.leaser, pc.locker, pc.accesses, *opts.AtimeBufferSize, *opts.MinEvictionAge, opts.Name)
+			pe, err := newPartitionEvictor(part, pc.fileStorer, blobDir, pc.leaser, pc.locker, pc, pc.accesses, *opts.AtimeBufferSize, *opts.MinEvictionAge, opts.Name)
```
Comment (maggie-lou) [inline enterprise/server/backends/pebble_cache/pebble_cache.go:380]: nit: Don't feel strongly about this but would it be cleaner to just pass in the pebble cache as an argument, as opposed to passing it in as a versionGetter interface + pass in a bunch of its other fields individually?

File: enterprise/server/backends/pebble_cache/pebble_cache.go [hunk 4]
```diff
@@ -1771,16 +1777,30 @@ func (e *partitionEvictor) Statusz(ctx context.Context) string {
...
 var digestChars = []byte("abcdef1234567890")
 
-func (e *partitionEvictor) randomKey(n int) []byte {
-	partID := e.partitionKeyPrefix()
-	cacheType := "/cas/"
-	buf := bytes.NewBuffer(make([]byte, 0, len(partID)+len(cacheType)+n))
-	buf.Write([]byte(partID))
-	buf.Write([]byte(cacheType))
-	for i := 0; i < n; i++ {
+func (e *partitionEvictor) randomKey(digestLength int) ([]byte, error) {
```
Comment (maggie-lou) [inline enterprise/server/backends/pebble_cache/pebble_cache.go:1780]: nit: Should this be a constant

File: enterprise/server/backends/pebble_cache/pebble_cache.go (hunk 3 above)
Comment (tylerwilliams) [reply enterprise/server/backends/pebble_cache/pebble_cache.go:380]: Yeah this bothered me too -- but pebble_cache holding evictors and evictors needing a reference to pebble cache seemed kinda gross. I'll prolly fix this in a followup.

File: enterprise/server/backends/pebble_cache/pebble_cache.go (hunk 4 above)
Comment (tylerwilliams) [reply enterprise/server/backends/pebble_cache/pebble_cache.go:1780]: It's this way to allow us to support multiple digests in the future.

**Pull request merged**